clang/lib/Serialization/ASTReaderDecl.cpp

```diff
@@ -1134,8 +1134,18 @@ void ASTDeclReader::VisitFunctionDecl(FunctionDecl *FD) {
   // the presence of a sycl_kernel_entry_point attribute, register it so that
   // associated metadata is recreated.
   if (FD->hasAttr<SYCLKernelEntryPointAttr>()) {
+    const auto *SKEPAttr = FD->getAttr<SYCLKernelEntryPointAttr>();
     ASTContext &C = Reader.getContext();
-    C.registerSYCLEntryPointFunction(FD);
+    const SYCLKernelInfo *SKI = C.findSYCLKernelInfo(SKEPAttr->getKernelName());
+    if (SKI) {
+      if (!declaresSameEntity(FD, SKI->getKernelEntryPointDecl())) {
+        Reader.Diag(FD->getLocation(), diag::err_sycl_kernel_name_conflict);
```

Comment:
These tend to be for modules specific stuff, and only as exceptions.  Though upon closer look, this might be for modules-related a diagnostic, but I'll have to take a closer look.

Reply:
The diagnostic being issued here is specifically for PCH/modules related scenarios. See the `sycl-kernel-entry-point-attr-kernel-name-module.cpp` and `sycl-kernel-entry-point-attr-kernel-name-pch.cpp` tests in `clang/test/SemaSYCL`.